### PR TITLE
Cancel an inference request when an error occurs

### PR DESCRIPTION
**Pull request by Ladicle**

### CODE REVIEW

File: engine/internal/runtime/manager.go [hunk 1]
```diff
@@ -290,16 +312,38 @@ func (m *Manager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result,
...
+			log.Error(err, "Failed to list pods")
+			return ctrl.Result{}, err
+		}
+		for _, pod := range podList.Items {
+			if pod.Labels["controller-revision-hash"] != sts.Status.CurrentRevision {
+				continue
+			}
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodScheduled {
+					if cond.Status == corev1.ConditionFalse &&
```
Comment (kkaneda) [inline engine/internal/runtime/manager.go:330]: Just wanted to make sure this works fine when a cluster is scaling up. The current request gets canceled, but a subsequent request will succeed once a pod is scheduled.

File: engine/internal/runtime/manager.go [hunk 2]
```diff
@@ -290,16 +312,38 @@ func (m *Manager) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result,
...
+		); err != nil {
+			log.Error(err, "Failed to list pods")
+			return ctrl.Result{}, err
+		}
+		for _, pod := range podList.Items {
+			if pod.Labels["controller-revision-hash"] != sts.Status.CurrentRevision {
+				continue
+			}
+			for _, cond := range pod.Status.Conditions {
+				if cond.Type == corev1.PodScheduled {
```
Comment (kkaneda) [inline engine/internal/runtime/manager.go:329]: No strong opinion, but we can combine these two `if`s into one.

**Review by kkaneda — APPROVED** [review]
LG!

File: engine/internal/runtime/manager.go (hunk 2 above)
Comment (Ladicle) [reply engine/internal/runtime/manager.go:329]: The first `if` is used for an early break.

File: engine/internal/runtime/manager.go (hunk 1 above)
Comment (Ladicle) [reply engine/internal/runtime/manager.go:330]: Yes, this only affects the waiting requests at that time, and subsequent requests do not change their behavior.

**Pull request merged**